ggshield/cmd/secret/scan/docset.py

```diff
@@ -0,0 +1,86 @@
+import json
+from typing import Callable, Iterable, Iterator, List
+
+import click
+
+from ggshield.core.constants import MAX_WORKERS
+from ggshield.core.text_utils import create_progress_bar
+from ggshield.core.utils import ScanContext, ScanMode, handle_exception
+from ggshield.output import OutputHandler
+from ggshield.scan import File, ScanCollection
+from ggshield.scan.scannable import Scanner
+
+
+def generate_files_from_docsets(path: str, verbose: bool = False) -> Iterator[File]:
+    with open(path, "r") as file:
+        for line in file:
+            obj = json.loads(line)
+            documents = obj["documents"]
+            for document in documents:
+                if verbose:
+                    click.echo(f"  * {document['id']}", err=True)
+                yield File(document["content"], document["id"])
+
+
+def create_scans_from_docset_paths(
+    scanner: Scanner,
+    paths: Iterable[str],
+    progress_callback: Callable[[], None],
+    verbose: bool = False,
+) -> List[ScanCollection]:
+    scans: List[ScanCollection] = []
+
+    for path in paths:
+        if verbose:
+            click.echo(f"- {click.format_filename(path)}", err=True)
+
+        files = generate_files_from_docsets(path, verbose)
+        results = scanner.scan(
+            files, scan_threads=MAX_WORKERS, progress_callback=lambda **x: None
+        )
+        scans.append(ScanCollection(id=path, type="docset", results=results))
+        progress_callback()
+
+    return scans
+
+
+@click.command()
+@click.argument(
+    "paths", nargs=-1, type=click.Path(exists=True, resolve_path=True), required=True
+)
+@click.pass_context
+def docset_cmd(ctx: click.Context, paths: List[str]) -> int:  # pragma: no cover
+    """
+    scan docset JSONL files.
```

Comment:
At some point we will have to publish a spec of the JSONL format.

Reply:
Indeed, maybe worth waiting until the doc stabilizes for sure?

Reply:
Yes, makes sense.